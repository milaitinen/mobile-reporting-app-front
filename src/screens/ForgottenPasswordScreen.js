import React from 'react';
import { StyleSheet, View, Text, TouchableHighlight } from 'react-native';

export default class ForgottenPasswordScreen extends React.Component {

  handleButtonPress = () => {
      this.props.navigation.goBack();
  };

  render() {
      return (
          <View style={styles.container}>
              <TouchableHighlight style={styles.button} onPress={() => this.handleButtonPress()}>
                  <Text style={styles.text}>Forgotten Password Screen</Text>
              </TouchableHighlight>
          </View>
      );
  }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },
    button: {
        backgroundColor: '#4F6D7A',
        width: 250,
        height: 50,
        justifyContent: 'center',
    },
    text: {
        textAlign: 'center',
        color: '#fff',
    }
});