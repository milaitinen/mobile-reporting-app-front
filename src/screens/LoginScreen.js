import React from 'react';
import { Text, StatusBar } from 'react-native';

import loginStyles from './style/loginStyles';
import { SignInButton } from '../components/Button';
import { Input } from '../components/TextInput';
import { AppBackground } from '../components/AppBackground';

export default class LoginScreen extends React.Component {

    constructor(props)
    {
        super(props);
        this.state = {
            isLoading: true,
            TextInputUser: '',
            TextInputPassword: '',
            TextInputServer: ''
        };
    }

    render() {
        return (
            <AppBackground>
                <StatusBar backgroundColor='#3d4f7c' barStyle='light-content'/>

                <Text style={loginStyles.title}>
                    MR-Application
                </Text>

                <Text style={loginStyles.slogan}>
                    Keep calm and keep reporting
                </Text>

                <Input
                    name={'user'}
                    placeholder='Email'
                    onChangeText={TextInputUser => this.setState({ TextInputUser })}
                />
                <Input
                    name={'lock'}
                    secureTextEntry={true}
                    placeholder='Password'
                    onChangeText={TextInputPassword => this.setState({ TextInputPassword })}
                />
                <Input
                    name={'globe'}
                    placeholder='Server url'
                    onChangeText={TextInputServer => this.setState({ TextInputServer })}
                />

                <SignInButton onPress={() => this.props.navigation.navigate('drawerStack')}>
                    Sign In
                </SignInButton>

                <Text style={loginStyles.copyright}>
                    Copyright Â© Arter Oy 2017
                </Text>
            </AppBackground>
        );
    }
}