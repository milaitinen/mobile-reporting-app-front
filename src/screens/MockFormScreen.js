import React from 'react';
import { View, FlatList, Text, StyleSheet } from 'react-native';

const dummyForms = [...new Array(12)].map((e, i) => `MockForm${i + 1}`);


export default class MockForm extends React.Component {
  static displayName = 'MockForm';

  render() {
      return (
          <View>
              <FlatList
                  data={dummyForms}
                  renderItem={({ item }) => <Text style={styles.FlatListItemStyle}>{item}</Text>}
                  keyExtractor={(item, index) => index}
              />
          </View>
      );
  }
}

const styles = StyleSheet.create({
    FlatListItemStyle: {
        padding: 10,
        fontSize: 15,
        height: 44
    }
});