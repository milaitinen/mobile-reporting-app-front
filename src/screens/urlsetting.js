export const url = 'http://aaltodev.arter.fi:8080/';
