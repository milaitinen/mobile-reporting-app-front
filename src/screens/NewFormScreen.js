import React from 'react';
import { StyleSheet, View, TextInput, Platform, Alert, Button } from 'react-native';
import { url } from './urlsetting';
import { NavigationActions } from 'react-navigation';
export default class NewFormScreen extends React.Component {


    constructor(props)
    {
        super(props);
        this.state = {
            TextInputName  : '',                                      // Text input is initialized as an empty string.
            layoutID       : props.navigation.state.params.layoutID,  /* LayoutID that is inherited from navigation
                                                                         parameters as stated in TemplateScreen class. */

        };

    }

    // Gets the current date and return it as a string.

    getDate = () => {

        const today = new Date();
        let dd = today.getDate();
        let mm = today.getMonth() + 1; // January is 0!
        const yyyy = today.getFullYear();

        if (dd < 10) {
            dd = '0' + dd;
        }

        if (mm < 10) {
            mm = '0' + mm;
        }

        return yyyy + '-' + mm + '-' + dd;
    }


    // Inserts data to server with a post method.

    InsertDataToServer = () => {

        const date = this.getDate();

        fetch(url + '/users/1/forms', {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },

            body: JSON.stringify({
                layoutID: this.state.layoutID,
                title: this.state.TextInputName,
                dateCreated: date,
                answers: [
                    {
                        fieldID: 1,
                        answer: "Answer 1"
                    },
                    {
                        fieldID: 2,
                        answer: "Answer 2"
                    }
                ]

            })

        }).then(response => {
            if (response.status === 200) {
                this.props.navigation.state.params.refresh();
                this.props.navigation.dispatch(NavigationActions.back());
                return 'Report added';
            } else {
                return response.status;
            }
        }).then((message) => {
            // Showing response message coming from server after inserting records.
            Alert.alert(message);

        }).catch((error) => {
            console.error(error);
        });

    }
    
    render() {
        return (
            <View style={styles.MainContainer}>
                <TextInput
                    placeholder='Enter Report Name'

                    onChangeText={TextInputName => this.setState({ TextInputName })}

                    underlineColorAndroid='transparent'

                    style={styles.TextInputStyleClass}
                />
                <Button
                    title='Create new report'
                    onPress={ this.InsertDataToServer }
                >
                </Button>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    MainContainer: {
        justifyContent: 'center',
        // alignItems: 'center',
        flex: 1,
        paddingTop: (Platform.OS === 'ios') ? 20 : 0,
    },

    TextInputStyleClass: {
        textAlign: 'center',
        marginBottom: 7,
        height: 40,
        borderWidth: 1,
        borderColor: '#FF5722',
    },
});