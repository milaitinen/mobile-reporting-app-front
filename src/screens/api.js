import { Platform, AsyncStorage, NetInfo } from 'react-native';

import { url } from './urlsetting';

export const login = (username, password) => {
    return fetch(`${url}/login`, {
        method: 'POST',
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json',
        },
        body: JSON.stringify({
            'username': username,
            'password': password,
        })
    }).then(response => {
        return JSON.parse(response._bodyInit).token;
    }).catch(err => alert(err));
};

// Send a new report to the server, along with the username and token.
export const createNewReport = (username, report, token) => {
    return fetch(`${url}/users/${username}/reports`, {
        method: 'POST',
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${token}`,
        },
        body: JSON.stringify(report)
    });
};

export const fetchFieldsByTemplateID = (username, templateID, token) => {
    return isNetworkConnected()
        .then((isConnected) => {
            if (!isConnected) { return fetchLocalFieldsByTemplateID(username, templateID); }
            return fetchRemoteFieldsByTemplateID(username, templateID, token);
        })
        .then((fieldsByTemplateID) => {
            saveData(`${url}/users/${username}/templates/${templateID}/fields`, fieldsByTemplateID);
            return fieldsByTemplateID;
        });
};

const fetchLocalFieldsByTemplateID = (username, templateID) => {
    return AsyncStorage.getItem(`${url}/users/${username}/templates/${templateID}/fields`)
        .then(data => {
            if (data !== null) {
                return JSON.parse(data);
            } else {
                return [];
            }
        });
};

export const fetchRemoteFieldsByTemplateID = (username, templateID, token) => {
    return (
        fetch(`${url}/users/${username}/templates/${templateID}/fields`, {
            headers: {
                'Authorization': `Bearer ${token}`,
            }
        })
            .then(response => {
                return response.json();
            })
    );
};

// used to store drafts
export const saveDraft = (username, templateID, draft) => {
    fetchDraftsByTemplateID(username, templateID)
        .then((drafts) => {
            // see if there is already a draft with the same id
            const draftIndex = drafts.findIndex(x => x.id === draft.id);

            if (draftIndex < 0) {
                drafts.push(draft);
            } else {
                drafts[draftIndex] = draft;
            }
            // give each draft a unique, negative id
            drafts.map((draft, i) => draft.id = -Math.abs(i + 1));
            saveData(`${url}/users/${username}/templates/${templateID}`, drafts);

            return (drafts[drafts.length - 1].id);
        });
};

export const removeDraft =  (username, templateID, draftID) => {
    fetchDraftsByTemplateID(username, templateID)
        .then((drafts) => {
            const draftRemoved = drafts.filter(draft => draft.id !== draftID);
            saveData(`${url}/users/${username}/templates/${templateID}`, draftRemoved);
        });
};

export const fetchDraftsByTemplateID = (username, templateID) => {
    return AsyncStorage.getItem(`${url}/users/${username}/templates/${templateID}`) // NOTE: this url is just a key
        .then(data => {
            if (data !== null) {
                return JSON.parse(data);
            } else {
                return [];
            }
        });
};

// fetch reports that are only locally saved
export const fetchLocalReports = (username, templateID) => {
    return AsyncStorage.getItem(`${url}/users/${username}/templates/${templateID}`) // NOTE: this url is just a key
        .then(data => {
            if (data !== null) {
                return JSON.parse(data);
            } else {
                return [];
            }
        });
};

// for future use?
export const fetchFieldsByReportID = (username, templateID, reportID, token) => {
    return isNetworkConnected()
        .then((isConnected) => {
            if (!isConnected || reportID === 0) { return fetchLocalFieldsByReportID(username, templateID, reportID); }
            return fetchRemoteFieldsByReportID(username, templateID, reportID, token);
        })
        .then((fieldsByReportID) => {
            saveData(`${url}/users/${username}/templates/${templateID}/reports/${reportID}/fields`, fieldsByReportID);
            return fieldsByReportID;
        });
};

// not yet in use
export const fetchLocalFieldsByReportID = (username, templateID, reportID) => {
    return AsyncStorage.getItem(`${url}/users/${username}/templates/${templateID}/reports/${reportID}/fields`)
        .then(data => {
            if (data !== null) {
                return JSON.parse(data);
            } else {
                return [];
            }
        });
};

// not yet in use
const fetchRemoteFieldsByReportID = (username, templateID, reportID, token) => {
    return (
        fetch(`${url}/users/${username}/templates/${templateID}/reports/${reportID}/fields`, {
            headers: {
                'Authorization': `Bearer ${token}`,
            }
        })
            .then(response => {
                return response.json();
            })
    );
};


/*
 Fetch templates from the server or ASyncStorage, depending on the availability of internet connection.
 Fetch templates that the user has rights to. */
export const fetchTemplatesByUsername = (username, token) => {
    return isNetworkConnected()
        .then((isConnected) => {
            if (!isConnected) { return fetchLocalTemplatesByUsername(username); }
            return fetchRemoteTemplatesByUsername(username, token);
        })
        .then((templates) => {
            saveData(`${url}/users/${username}/templates`, templates);
            return templates;
        });
};

/*
 Fetch all templates from ASyncStorage in case there is no internet connection.
 If no data has been stored an empty value will be returned. */
const fetchLocalTemplatesByUsername = (username) => {
    return AsyncStorage.getItem(`${url}/users/${username}/templates`)
        .then(data => {
            if (data !== null) {
                return JSON.parse(data);
            } else {
                return [];
            }
        });
};

// Fetch all templates from the server
const fetchRemoteTemplatesByUsername = (username, token) => {
    return (
        fetch(`${url}/users/${username}/templates`, {
            headers: {
                'Authorization': `Bearer ${token}`
            }
        })
            .then(response => {
                if (response._bodyText === '') response._bodyText = '[]';
                return response.json();
            })
    );
};

/* Fetch reports by their templateID from the server if online, ASyncStorage otherwise */
export const fetchReportsByTemplateID = (username, templateID, token) => {
    return isNetworkConnected()
        .then((isConnected) => {
            if (!isConnected) { return fetchLocalReportsByTemplateID(username, templateID); }
            return fetchRemoteReportsByTemplateID(username, templateID, token);
        })
        .then((reports) => {
            saveData(`${url}/users/${username}/templates/${templateID}/reports?sort=-datecreated`, reports);
            return reports;
        });
};

/* Fetch Reports by TemplateID from ASyncStorage in case there is no internet connection.
   If no data has been stored an empty value will be returned. */
const fetchLocalReportsByTemplateID = (username, templateID) => {
    return AsyncStorage.getItem(`${url}/users/${username}/templates/${templateID}/reports?sort=-datecreated`)
        .then(data => {
            if (data !== null) {
                return JSON.parse(data);
            } else {
                return [];
            }
        });
};

// Fetch reports by templateID from the server
const fetchRemoteReportsByTemplateID = (username, templateID, token) => {
    return (
        fetch(`${url}/users/${username}/templates/${templateID}/reports?sort=-datecreated`, {
            headers: {
                'Authorization': `Bearer ${token}`
            }
        })
            .then(response => {
                return response.json();
            })
    );
};

/* Returns all the reports made by the user specified by the username. */
export const fetchReportsByUsername = (username, token) => {
    return isNetworkConnected()
        .then((isConnected) => {
            if (!isConnected) { return fetchLocalReportsByUsername(username); }
            return fetchRemoteReportsByUsername(username, token);
        })
        .then((reports) => {
            saveData(`${url}/users/${username}/reports`, reports);
            return reports;
        });
};

const fetchLocalReportsByUsername = (username) => {
    return AsyncStorage.getItem(`${url}/users/${username}/reports`)
        .then(data => {
            if (data !== null) {
                return JSON.parse(data);
            } else {
                return [];
            }
        });
};

const fetchRemoteReportsByUsername = (username, token) => {
    return (
        fetch(`${url}/users/${username}/reports`, {
            headers: {
                'Authorization': `Bearer ${token}`
            }
        })
            .then(response => {
                return response.json();
            })
    );
};

/* Store data (layouts, reports - depending on the url) to ASyncStorage, a simple key-value storage system global to the app.
   Keys and values are stored as a string. */
const saveData = (dataUrl, data) => {
    AsyncStorage.setItem(dataUrl, JSON.stringify(data));
};

/*
const removeData = (dataUrl) => {
    try {
        AsyncStorage.removeItem(dataUrl);
    } catch (error) {
        console.error(error);
    }
};
*/

// Necessary because of a bug on iOS https://github.com/facebook/react-native/issues/8615#issuecomment-287977178
export function isNetworkConnected() {
    if (Platform.OS === 'ios') {
        return new Promise(resolve => {
            const handleFirstConnectivityChangeIOS = isConnected => {
                NetInfo.isConnected.removeEventListener('connectionChange', handleFirstConnectivityChangeIOS);
                resolve(isConnected);
            };
            NetInfo.isConnected.addEventListener('connectionChange', handleFirstConnectivityChangeIOS);
        });
    }

    return NetInfo.isConnected.fetch();
}
