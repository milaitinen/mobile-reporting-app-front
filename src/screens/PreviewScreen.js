import React from 'react';
import { View, ScrollView, TextInput, Text, ActivityIndicator, Linking } from 'react-native';
import { Icon } from 'react-native-elements';
import { Checkbox } from '../components/Checkbox';
import { Dropdown } from '../components/Dropdown';
import RadioForm, { RadioButton, RadioButtonInput, RadioButtonLabel } from 'react-native-simple-radio-button';
import DatePicker from 'react-native-datepicker';
import ModalDropdown from 'react-native-modal-dropdown';
import { connect } from 'react-redux';

import { AppBackground } from '../components/AppBackground';
import { EditButton } from '../components/EditButton';
import { fetchFieldsByTemplateID } from './api';
import { strings } from '../locales/i18n';
import { insertTitle } from '../redux/actions/preview';
import { createReport } from '../redux/actions/newReport';

import newReportStyles from './style/newReportStyles';
import templateScreenStyles from './style/templateScreenStyles';
import styles from '../components/Dropdown/styles';

// "export" necessary in order to test component without Redux store
export class PreviewScreen extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isLoading      : true,
            number         : '',
        };
    }

    componentDidMount() {
        this.getFieldsByID();
    }

    getFieldsByID = () => {
        fetchFieldsByTemplateID(this.props.username, this.props.templateID, this.props.token)
            .then(responseJson => {
                this.setState({ dataFieldsByID: responseJson, isLoading: false });
            })
            .catch(error => console.error(error) )
            .done();
    };

    handleOnPress = () => {
        this.props.dispatch(createReport(this.props.templateID, true));
        this.props.navigation.navigate('NewReport', { refresh: this.handleRefresh });
    };

    onChanged = (text) => {
        let newText = '';
        const numbers = '0123456789';

        for (let i=0; i < text.length; i++) {
            if (numbers.indexOf(text[i]) > -1 ) {
                newText = newText + text[i];
            } else {
                alert('Please enter numbers only');
            }
        }
        this.setState({ number: newText });
    };

    render() {

        if (this.state.isLoading) {
            return (
                <AppBackground>
                    <ActivityIndicator
                        animating={this.state.animating}
                        style={[templateScreenStyles.activityIndicator, { height: 80 }]}
                        size='large'
                        color={'#9dcbe5'}
                    />
                </AppBackground>
            );
        }

        const { isEditable } = this.props;
        const renderedFields = this.state.dataFieldsByID.map((field, index) => {
            switch (field.typeID) {

                case 1: // Name
                    return (
                        <View key={index}>
                            <Text style={newReportStyles.textStyleClass}>{field.title}</Text>
                            <TextInput
                                editable={isEditable}
                                placeholder={field.defaultValue}
                                placeholderTextColor={'#adadad'}
                                onSubmitEditing={(event) => this.props.dispatch(insertTitle(event.nativeEvent.text))}
                                underlineColorAndroid='transparent'
                                style={newReportStyles.textInputStyleClass}
                            />
                        </View>
                    );

                case 2: // Checkbox
                    return (
                        <View key={index}>
                            <Text style={newReportStyles.textStyleClass}>{field.title}</Text>
                            <Checkbox
                                key={index}
                                style={ newReportStyles.checkboxStyle }
                                title={'This is a nice checkbox'}
                                editable={isEditable}
                            />
                        </View>
                    );

                case 3: // Dropdown
                    return (
                        <View key={index}>
                            <Text style={ newReportStyles.textStyleClass }>{field.title}</Text>
                            <ModalDropdown
                                disabled={!isEditable}
                                options={['option 1', 'option 2']}
                                dropdownStyle={ newReportStyles.dropStyleClass }
                                defaultValue={'Select option'}
                                style={newReportStyles.dropdownButton}
                                textStyle={newReportStyles.dropdownText}
                                renderRow={ () =>
                                    <View>
                                        <ModalDropdown
                                            options={['option 3', 'option 4']}
                                            style={ newReportStyles.lowerDropdownStyleClass }
                                            dropdownStyle={ styles.dropStyleClass }
                                        />
                                    </View>
                                }
                            >
                                <View style={styles.buttonContent}>
                                    <Text style={styles.dropdownText}>
                                        Select option
                                    </Text>
                                    <Icon name={'expand-more'} color={'#adadad'} style={styles.icon}/>
                                </View>
                            </ModalDropdown>
                        </View>

                    );

                case 4: // TextRow (One row text field)
                    return (
                        <View key={index}>
                            <Text style={ newReportStyles.textStyleClass }>{field.title}</Text>
                            <TextInput
                                editable={isEditable}
                                placeholder={field.defaultValue}
                                placeholderTextColor={'#adadad'}
                                underlineColorAndroid='transparent'
                                style={newReportStyles.textInputStyleClass}
                            />
                        </View>
                    );

                case 5: // Choice (Yes/No) NOTE: Error will be removed when options come from the database.
                    const props = [{ label: 'Yes', value: 1 }, { label: 'No', value: 0 }];
                    return (
                        <View key={index}>
                            <Text style={ newReportStyles.textStyleClass }>{field.title}</Text>
                            <View style={ newReportStyles.radioButtonContainer }>
                                <RadioForm
                                    animation={true}
                                    formHorizontal={true}
                                >
                                    {props.map((obj, i) =>
                                        <RadioButton
                                            key={i}
                                        >
                                            <RadioButtonInput
                                                disabled={!isEditable}
                                                obj={obj}
                                                index={i}
                                                isSelected={this.state.value === obj.value}
                                                onPress={(value) => { this.setState({ value: value }); }}
                                                borderWidth={1}
                                                buttonColor={'#8cc9e5'}
                                                buttonSize={16}
                                                buttonOuterSize={24}
                                                buttonStyle={{}}
                                                buttonWrapStyle={{}}
                                            />
                                            <RadioButtonLabel
                                                disabled={!isEditable}
                                                obj={obj}
                                                index={i}
                                                labelHorizontal={true}
                                                onPress={(value) => { this.setState({ value: value }); }}
                                                labelStyle={{ marginRight: 20 }}
                                                labelWrapStyle={{}}
                                            />
                                        </RadioButton>
                                    )}
                                </RadioForm>
                            </View>
                        </View>
                    );

                case 6: // Calendar
                    return (
                        <View key={index} >
                            <Text style={ newReportStyles.textStyleClass }>{field.title}</Text>
                            <DatePicker
                                disabled={!isEditable}
                                style={ newReportStyles.dateStyleClass }
                                customStyles={{
                                    dateInput: {
                                        borderColor: '#adadad',
                                        borderWidth: 1.5,
                                        borderRadius: 5,
                                        backgroundColor: 'white',
                                    },
                                    dateText: {
                                        color: '#adadad',
                                    }
                                }}
                                date={field.defaultValue}
                                mode="date"
                                placeholder="select date"
                                placeholderTextColor={'#adadad'}
                                format="YYYY-MM-DD"
                                minDate="2018-05-01"
                                maxDate="2018-06-01"
                                confirmBtnText="Confirm"
                                cancelBtnText="Cancel"
                                /*iconComponent={<Icon name={'event'} type={'MaterialIcons'} iconStyle={ newReportStyles.dateIconStyle }/>}*/
                                onDateChange={(date) => {this.setState({ date: date });}}
                            />
                        </View>
                    );

                case 7: // Instructions
                    return (
                        <View key={index}>
                            <Text style={ newReportStyles.textStyleClass }>{field.title}</Text>
                            <Text
                                style = {newReportStyles.instructions}>
                                {field.defaultValue}
                            </Text>
                        </View>
                    );

                case 8: // Text (Multiple row text field)
                    return (
                        <View key={index}>
                            <Text style={ newReportStyles.textStyleClass }>{field.title}</Text>
                            <TextInput
                                editable={isEditable}
                                style = {newReportStyles.multilinedTextInputStyleClass}
                                placeholder={field.defaultValue}
                                placeholderTextColor={'#adadad'}
                                multiline={true}
                            />
                        </View>
                    );

                case 9: // Time
                    return (
                        <View key={index}>
                            <Text style={ newReportStyles.textStyleClass }>{field.title}</Text>
                            <DatePicker
                                disabled={!isEditable}
                                style={ newReportStyles.dateStyleClass }
                                customStyles={{
                                    dateInput: {
                                        borderColor: '#adadad',
                                        borderWidth: 1.5,
                                        borderRadius: 5,
                                        backgroundColor: 'white',
                                    },
                                    dateText: {
                                        color: '#adadad',
                                    }
                                }}
                                date={field.defaultValue}
                                mode="time"
                                format="HH:mm"
                                confirmBtnText="Confirm"
                                cancelBtnText="Cancel"
                                minuteInterval={10}
                                iconComponent={<Icon name={'clock'} type={'entypo'} iconStyle={ newReportStyles.dateIconStyle }/>}
                                onDateChange={(time) => {this.setState({ time: time });}}
                            />
                        </View>
                    );

                case 10: // Digits (Text input that only accepts numeric characters)
                    return (
                        <View key={index}>
                            <Text style={ newReportStyles.textStyleClass }>{field.title}</Text>
                            <TextInput
                                editable={isEditable}
                                style={newReportStyles.textInputStyleClass}
                                placeholder={field.defaultValue}
                                placeholderTextColor={'#adadad'}
                                keyboardType = 'numeric'
                                onChangeText={(text)=> this.onChanged(text)}
                                value = {this.state.number}
                            />
                        </View>
                    );

                case 11: // Link
                    return (
                        <View key={index} style={{ flexDirection: 'row' }}>
                            <Icon name={'link'} type={'feather'} iconStyle={ newReportStyles.linkIconStyle }/>
                            <Text
                                disabled={!isEditable}
                                style={ newReportStyles.linkStyleClass }
                                onPress={() => Linking.openURL(field.defaultValue)}>
                                Link to somewhere
                            </Text>
                        </View>
                    );

                case 12: // User dropdown
                    return (
                        <View key={index}>
                            <Text style={ newReportStyles.textStyleClass }>{field.title}</Text>
                            <Dropdown
                                disabled={!isEditable}
                                defaultValue={'Select user'}
                                options={JSON.parse(field.defaultValue)}
                            />
                        </View>
                    );

                default:
                    return (
                        null
                    );

            }
        });

        return (
            <AppBackground style={'no-padding'}>
                <View style={ newReportStyles.ViewContainer }>
                    <View style={ newReportStyles.ReportContainer }>
                        <ScrollView keyboardShouldPersistTaps={'handled'} style={ newReportStyles.ReportScrollView }>
                            <View style={newReportStyles.titleContainer}>
                                <Icon name={'assignment'} color={'#a0a0a0'} size={45}/>
                                <Text style={newReportStyles.title}>{strings('templates.report')}</Text>
                                <Icon name={'find-in-page'} color={'#88c9e5'} size={45} containerStyle={newReportStyles.previewIconContainer}/>
                            </View>
                            <View style={newReportStyles.fieldContainer}>
                                {renderedFields}
                            </View>
                        </ScrollView>
                        <EditButton onPress={() => this.handleOnPress()} />
                    </View>
                </View>
            </AppBackground>
            /*
                <TextInput
                    placeholder={ strings('createNew.enterNew') }
                    onChangeText={(TextInputName) => this.setState({ TextInputName })}
                    underlineColorAndroid='transparent'
                    style={newReportStyles.textInputStyleClass}
                />
                <Button
                    title={ strings('createNew.createNew') }
                    onPress={ this.send }
                >
                </Button>
            */
        );
    }
}

// maps redux state to component props. Object that is returned can be accessed via 'this.props' e.g. this.props.email
const mapStateToProps = (state) => {
    const userID = state.user.userID;
    const isEditable = state.preview.isEditable;
    const templateID = state.preview.templateID;
    const title = state.preview.title;
    const number = state.preview.number;
    const token = state.user.token;
    const username = state.user.username;
    return {
        userID,
        isEditable,
        templateID,
        title,
        number,
        token,
        username
    };
};

export default connect(mapStateToProps)(PreviewScreen);