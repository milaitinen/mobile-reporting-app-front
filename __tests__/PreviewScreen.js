import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'isomorphic-fetch';
import { shallow } from 'enzyme';
import { ActivityIndicator } from 'react-native';

import { PreviewScreen } from '../src/screens/PreviewScreen';
import { url } from '../src/screens/urlsetting';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
    const tree = renderer.create(
        <PreviewScreen />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('fetch finds data from the server', () => {
    const data = fetch(url + '/users/1/forms');
    const userData = fetch(url + '/users/1');

    expect(userData).not.toBe(null);
    expect(data).not.toBe(null);
});

describe('<PreviewScreen />', () => {
    describe('isLoading', () => {

        const previewScreen = shallow(<PreviewScreen />);

        it('should render an <ActivityIndicator /> if true', () => {
            previewScreen.setState({ isLoading: true });
            expect(previewScreen.find(ActivityIndicator).length).toBe(1);
        });

        it('should not render an <ActivityIndicator /> if false', () => {
            previewScreen.setState({ isLoading: false, dataFieldsByID: [] });
            expect(previewScreen.find(ActivityIndicator).length).toBe(0);
        });
    });
});