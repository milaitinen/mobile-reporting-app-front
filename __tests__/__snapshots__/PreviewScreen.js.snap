// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  colors={
    Array [
      undefined,
      undefined,
      undefined,
    ]
  }
  style={undefined}
>
  <ActivityIndicator
    animating={true}
    color="#9dcbe5"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        undefined,
        Object {
          "height": 80,
        },
      ]
    }
  />
</View>
`;
